editor.planx.uk/src/pages/FlowEditor/components/FeedbackLog/components/ChangeStatusTool.tsx

```diff
@@ -0,0 +1,106 @@
+import RateReview from "@mui/icons-material/RateReview";
+import Button from "@mui/material/Button";
+import Menu from "@mui/material/Menu";
+import MenuItem from "@mui/material/MenuItem";
+import { styled, useTheme } from "@mui/material/styles";
+import {
+  gridRowSelectionStateSelector,
+  useGridApiContext,
+  useGridSelector,
+} from "@mui/x-data-grid";
+import * as React from "react";
+
+import { statusOptions } from "../feedbackFilterOptions";
+import { updateFeedbackStatus } from "../queries/updateFeedbackStatus";
+import { FeedbackStatus } from "../types";
+import { FEEDBACK_COLOURS } from "./StatusChip";
+
+const StyledMenuItem = styled(MenuItem, {
+  shouldForwardProp: (prop) => prop !== "color",
+})(({ theme, color }) => ({
+  borderLeft: `6px solid ${color}`,
+  marginLeft: theme.spacing(1),
+  paddingTop: theme.spacing(0.5),
+  paddingBottom: theme.spacing(0.5),
+  marginBottom: theme.spacing(1),
+  "&:last-child": {
+    marginBottom: 0,
+  },
+}));
+
+export const ChangeStatusTool: React.FC = () => {
+  const apiRef = useGridApiContext();
```

Comment:
Cool to see that data grid has built in / preconfigured context that we can hook into :heavy_plus_sign:

Reply:
Totally - this was a real nice realisation for me also - a whole load of built in functionality we can use out of the box.